core/src/worker/mod.rs

```diff
@@ -121,8 +123,13 @@ pub struct Worker {
     local_activities_complete: Arc<AtomicBool>,
     /// Used to track all permits have been released
     all_permits_tracker: tokio::sync::Mutex<AllPermitsTracker>,
-    /// Used to shutdown the worker heartbeat task
-    worker_heartbeat: Option<WorkerHeartbeatManager>,
+    /// Instance key used to identify this worker in worker heartbeating
+    worker_instance_key: String,
+    /// Collects data to send to server via worker heartbeat
+    worker_heartbeat_callback: Option<HeartbeatFn>,
+    /// Used to remove this worker from the parent map used to track this worker for
+    /// worker heartbeat
+    worker_heartbeat_shutdown_callback: OnceCell<Arc<dyn Fn() + Send + Sync>>,
```

Comment:
Probably would be nice to wrap these up in a little struct (can just be private to this file)